provisioning/baremetal_crypto.go

```diff
@@ -0,0 +1,104 @@
+/*
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package provisioning
+
+import (
+	"crypto/rand"
+	"math/big"
+	"time"
+
+	"k8s.io/apimachinery/pkg/util/sets"
+	"k8s.io/client-go/util/cert"
+
+	"github.com/openshift/library-go/pkg/crypto"
+)
+
+type TlsCertificate struct {
+	privateKey  string
+	certificate string
+}
+
+const (
+	tlsExpirationDays = 365 * 2
+	tlsRefreshDays    = 180
+)
+
+func generateRandomPassword() (string, error) {
+	chars := []rune("ABCDEFGHIJKLMNOPQRSTUVWXYZ" +
+		"abcdefghijklmnopqrstuvwxyz" +
+		"0123456789")
+	length := 16
+	buf := make([]rune, length)
+	numChars := big.NewInt(int64(len(chars)))
+	for i := range buf {
+		c, err := rand.Int(rand.Reader, numChars)
+		if err != nil {
+			return "", err
+		}
+		buf[i] = chars[c.Uint64()]
+	}
+	return string(buf), nil
+}
+
+func generateTlsCertificate(provisioningIP string) (TlsCertificate, error) {
+	caConfig, err := crypto.MakeSelfSignedCAConfig("metal3-ironic", tlsExpirationDays)
```

Comment:
The cluster has a CA. Should we use that to sign the certificate?

Reply:
Interesting, how do I access that?

Reply:
It doesn't look like there's a direct way, and the API that is exposed via k8s appears focused on client access certificates rather than general server certificates. That API requires an "approver" to work as part of the process, which we may not have when we need our certificate. Let's stick with self-signed for now and switch if we learn that we can.